File: ejemploFuncion2.cpp
#include <iostream>
using namespace std;
bool esPrimo(int numero);
void evaluarCondicion(int x, int y);
int main(){
    int a, b;
    cout << "Ingrese primer numero: ";
    cin >> a;
    cout << "Ingrese segundo numero: ";
    cin >> b;
    evaluarCondicion(a, b);
    return -100; 
}
bool esPrimo(int numero) {
    for (int i = 2; i < numero; i++)
    {
        if (numero % i == 0) {
            return false;
        }
    }
    return true;
}

void evaluarCondicion(int x, int y){
    if (esPrimo(x) && esPrimo(y)) {
        cout << "La suma es: " << x + y << endl;
    } else {
        cout << "No se cumple la condicion\n";
    }
}

File: ejemploFuncion2.py
#solicite al usuario dos numeros y sume si y solo si ambos sean primos
def esPrimo(numero):
    for i in range(2, numero):
        if numero % i == 0:
            print('El numero ', numero, ' no es primo, lo divide el numero ', i)
            return False
    return True

def evaluarCondicion(n1, n2):
    if esPrimo(a) & esPrimo(b):
        print('La suma es: ', a + b)
    else:
        print('No se cumple la condicion')

a = int(input('Ingrese un numero '))
b = int(input('Ingrese un numero '))
evaluarCondicion(a, b)